posix/posix.c

```diff
@@ -1068,68 +1071,68 @@ int posix_fstat(int fd, struct stat *buf)
 {
 	TRACE("fstat(%d)", fd);
 
-	msg_t msg;
 	open_file_t *f;
+	msg_t msg;
 	int err;
 
-	if (!(err = posix_getOpenFile(fd, &f))) {
-		hal_memset(buf, 0, sizeof(struct stat));
-		hal_memset(&msg, 0, sizeof(msg_t));
+	if ((err = posix_getOpenFile(fd, &f)) < 0)
+		return err;
 
-		buf->st_dev = f->ln.port;
-		buf->st_ino = (int)f->ln.id; /* FIXME */
-		buf->st_rdev = f->oid.port;
+	hal_memset(buf, 0, sizeof(struct stat));
```

Comment:
We should at least check if stat != NULL (we also really need to sanitize pointers passed to the kernel, i.e. check if they belong to the user address space, but thats for later)

Reply:
I'm leaving adding the checks for another PR.